dmscripts/generate_supplier_user_csv.py

```diff
@@ -0,0 +1,152 @@
+import csv
+from dmutils.s3 import S3ResponseError
+
+from dmscripts.models.writecsv import csv_path
+
+
+def generate_supplier_csv(framework_slug, data_api_client, logger):
+    framework = data_api_client.get_framework(framework_slug).get("frameworks")
+
+    supplier_rows = data_api_client.export_suppliers(framework_slug).get('suppliers', [])
+    if not supplier_rows:
+        logger.info('No supplier data found for framework {}'.format(framework_slug))
+        return [], [], None
+
+    supplier_and_framework_headers = [
+        "supplier_id",
+        "supplier_name",
+        "supplier_organisation_size",
+        "duns_number",
+        "companies_house_number",
+        "registered_name",
+        "declaration_status",
+        "application_status",
+        "application_result",
+        "framework_agreement",
+        "variations_agreed",
+    ]
+    service_count_headers = [lot['slug'] for lot in framework['lots']]
+    contact_info_headers = [
+        'contact_name',
+        'contact_email',
+        'contact_phone_number',
+        'address_first_line',
+        'address_city',
+        'address_postcode',
+        'address_country',
+    ]
+
+    filename = "official-details-for-suppliers-{}".format(framework_slug)
```

Comment:
Is it worth putting a datestamp in the filename?

Reply:
I'm a bit torn on that. The current report doesn't have a datestamp, but it's generated on demand so the user should 'know' its date. Adding it might be useful to flag up when the data has changed. On the other hand, it will make finding the file in the bucket and generating the URL more annoying...